From 7820011cc39012bf83af637e3c585872724319b2 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 9 May 2024 17:12:36 +0200
Subject: [PATCH 5/5] net: sfp: fix build error

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/net/phy/sfp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/sfp.c b/drivers/net/phy/sfp.c
index 4ad294b7c51b..123dc3f0e7c0 100644
--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -303,7 +303,7 @@ static int sfp_i2c_read(struct sfp *sfp, bool a2, u8 dev_addr, void *buf,
 
 		msgs[1].len = this_len;
 
-		ret = i2c_transfer(i2c, msgs, ARRAY_SIZE(msgs));
+		ret = i2c_transfer(sfp->i2c, msgs, ARRAY_SIZE(msgs));
 		if (ret < 0)
 			return ret;
 
-- 
2.35.3

